### DISCUSSION TITLE
Random Color Clock

### DISCUSSION THREAD

**Discussion by o0shojo0o** [pixelit-project/PixelIt#46] [Node-RED Flows]
**by dd_internetagentur**

Anzeige der Uhrzeit / Datum bei jedem Aufruf in einer zufälligen Farbe.
![image](https://user-images.githubusercontent.com/19333515/121051622-e03e7980-c7b9-11eb-9277-73444d0b460c.png)

~~~
[{"id":"75879130.9ea528","type":"inject","z":"bd011d21.f651b8","name":"CLOCK","topic":"","payload":"","payloadType":"date","repeat":"10","crontab":"","once":true,"onceDelay":"0.1","x":230,"y":270,"wires":[["4c162ae7.80828c"]]},{"id":"4c162ae7.80828c","type":"function","z":"bd011d21.f651b8","name":"RND Color","func":"var randomColor1 = Math.floor((Math.random() * 255) + 1);\nvar randomColor2 = Math.floor((Math.random() * 255) + 1);\nvar randomColor3 = Math.floor((Math.random() * 255) + 1);\n\nmsg.red     = randomColor1;\nmsg.green   = randomColor2;\nmsg.blue    = randomColor3;\n\nreturn msg;","outputs":1,"noerr":0,"x":380,"y":270,"wires":[["49b79ff.8b626e"]]},{"id":"c01e3b2c.2b4728","type":"Screen Init","z":"bd011d21.f651b8","name":"Screen Init","inscreenName":"internal_clock","induration":"8","x":740,"y":270,"wires":[["a89d9934.40ec78"]]},{"id":"49b79ff.8b626e","type":"Switch Animation","z":"bd011d21.f651b8","name":"Switch Animation","aktiv":"true","animation":"fade","x":560,"y":270,"wires":[["c01e3b2c.2b4728"]]},{"id":"a89d9934.40ec78","type":"Clock","z":"bd011d21.f651b8","name":"Clock","inswitchAktiv":"true","inswitchSec":"5","inwithSeconds":"true","incolorR":"{{red}}","incolorG":"{{green}}","incolorB":"{{blue}}","x":880,"y":270,"wires":[[]]}]
~~~

**Comment by maverik80** [comment]
hi how do i pass a color chosen by me in nodered? do you have an example?

**Reply by o0shojo0o** [reply]
In the clock node you can defer a color.
![image](https://user-images.githubusercontent.com/19333515/160761644-528f90f7-03fa-435c-81b2-5af968b50b74.png)

**Reply by maverik80** [reply]
I thank you for the answer, but maybe I expressed myself badly, I created an input_text from HA that sends me the desired text on the display and it works, but I would like to be able to change the color of that text and I can't

**Reply by davipet1** [reply]
double click the "text" node. You can adjust the R G B there.

**Reply by maverik80** [reply]
ok but I want it to be dynamic that by creating 3 entities from Home assistant I give it 3 values ​​that I pass on nodered while the text is inserted ... I hope I have explained. sorry my english

**Reply by davipet1** [reply]
OK, Got it.
Sorry, I am a noob on this myself. But I guess you can use 3 Inputs and a "join" node to pack them up to one msg.
On the text node you use the payload of the different topics to your R G B values serperatly.

**Reply by davipet1** [reply]
Import this in node red. I hope this helps...

`[
    {
        "id": "e436b366a923a779",
        "type": "ioBroker in",
        "z": "77f39cc6d8e46378",
        "name": "R Value",
        "topic": "0_userdata.0.R",
        "payloadType": "value",
        "onlyack": "",
        "func": "all",
        "gap": "",
        "fireOnStart": "true",
        "outFormat": "MQTT",
        "x": 90,
        "y": 1260,
        "wires": [
            [
                "54dad18a1b48d27e"
            ]
        ]
    },
    {
        "id": "332c62a92f3fef22",
        "type": "ioBroker in",
        "z": "77f39cc6d8e46378",
        "name": "G Value",
        "topic": "0_userdata.0.G",
        "payloadType": "value",
        "onlyack": "",
        "func": "all",
        "gap": "",
        "fireOnStart": "true",
        "outFormat": "MQTT",
        "x": 90,
        "y": 1300,
        "wires": [
            [
                "54dad18a1b48d27e"
            ]
        ]
    },
    {
        "id": "dc5db80debf1aa51",
        "type": "ioBroker in",
        "z": "77f39cc6d8e46378",
        "name": "B Value",
        "topic": "0_userdata.0.B",
        "payloadType": "value",
        "onlyack": "",
        "func": "all",
        "gap": "",
        "fireOnStart": "true",
        "outFormat": "MQTT",
        "x": 90,
        "y": 1340,
        "wires": [
            [
                "54dad18a1b48d27e"
            ]
        ]
    },
    {
        "id": "54dad18a1b48d27e",
        "type": "join",
        "z": "77f39cc6d8e46378",
        "name": "",
        "mode": "custom",
        "build": "object",
        "property": "payload",
        "propertyType": "msg",
        "key": "topic",
        "joiner": "\\n",
        "joinerType": "str",
        "accumulate": true,
        "timeout": "1",
        "count": "1",
        "reduceRight": false,
        "reduceExp": "",
        "reduceInit": "",
        "reduceInitType": "",
        "reduceFixup": "",
        "x": 310,
        "y": 1300,
        "wires": [
            [
                "728ab45a0bb338ee"
            ]
        ]
    },
    {
        "id": "16ee83cf48ae87e1",
        "type": "Switch Animation",
        "z": "77f39cc6d8e46378",
        "name": "Switch Animation",
        "aktiv": "true",
        "animation": "fade",
        "indata": "",
        "inwidth": "",
        "incolorHex": "",
        "incolorR": "",
        "incolorG": "",
        "incolorB": "",
        "x": 770,
        "y": 1300,
        "wires": [
            [
                "0232240982f66c5f"
            ]
        ]
    },
    {
        "id": "0232240982f66c5f",
        "type": "Text",
        "z": "77f39cc6d8e46378",
        "name": "Text",
        "intextString": "--Your Text--",
        "inbigFont": "false",
        "inscrollText": "false",
        "inscrollTextDelay": "0",
        "incenterText": "true",
        "inposX": "8",
        "inposY": "2",
        "incolorHex": "",
        "incolorR": "{{payload.0_userdata/0/R}}",
        "incolorG": "{{payload.0_userdata/0/G}}",
        "incolorB": "{{payload.0_userdata/0/B}}",
        "x": 970,
        "y": 1300,
        "wires": [
            []
        ]
    },
    {
        "id": "728ab45a0bb338ee",
        "type": "Bitmap Animation",
        "z": "77f39cc6d8e46378",
        "name": "Bitmap Animation",
        "indata": "123",
        "inanimationDelay": "100",
        "inrubberbanding": "false",
        "inlimitLoops": "0",
        "x": 550,
        "y": 1300,
        "wires": [
            [
                "16ee83cf48ae87e1"
            ]
        ]
    }
]`

**Reply by maverik80** [reply]
unfortunately I don't know how to use ioBroker and I don't know how to manage it, thanks for the answer, on home assistant I created 3 entities input_number Red, Green, Blue  from 0 to 255 but I can't send the three values ​​together on the core from nodered and therefore I can't change the main color of the led matrix

**Reply by o0shojo0o** [reply]
But you can also work with the HEX values!

**Reply by maverik80** [reply]
I have already tried to use HEx but I only receive a restart of the core, for crash I suppose .. in fact I do not understand how it is possible to send 3 numerical values ​​at the same time to change color to the LEDs

**Reply by maverik80** [reply]
I was able to easily adjust the brightness by connecting it to the matrix controll and then through automation in the home assistant I manage the brightness without a sensor but to my liking. but absurdly I can't set the main color of the text
